### PR TITLE
Add contributing guide

### PR DESCRIPTION
**Pull request by heitorPB**

### CODE REVIEW

**Review by zsinx6 — APPROVED** [review]
Very nice :+1:

File: contributing.rst [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+Code
+====
+
+OSD source-code is hosted in GitHub repositories. All the charms are in
+`omnivector-solutions/slurm-charms <https://github.com/omnivector-solutions/slurm-charms>`_,
+and a common code shared among them, called ``slurm-ops-mananger``, is hosted at
+`omnivector-solutions/slurm-ops-manager <https://github.com/omnivector-solutions/slurm-ops-manager>`_.
+
+By contributing with code development, you agree the license used.
```
Comment (jamesbeedy) [inline contributing.rst:89]:
```suggestion
By contributing to the slurm-charms project or affiliated code you are implicitly agreeing to the license used, MIT.
```

Comment (heitorPB) [reply contributing.rst]: We should also have a `Legal` section in the docs, what do you think?

**Comment by heitorPB** [comment]
Online at https://omnivector-solutions.github.io/osd-documentation/testing/contributing.html

**Pull request merged**